cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -23,6 +23,260 @@ Cask Data Application Platform Release Notes
+  Added a new Javascript transform that can emit records using emitter. Have deprecated Script transform.
+
+- `CDAP-4251 <https://issues.cask.co/browse/CDAP-4251>`__ -
+  Removed deprecated fluent style API for Flow configuration. The only supported API is now the configurer style.
+
+- `CDAP-2303 <https://issues.cask.co/browse/CDAP-2303>`__ -
+  Added lookup capability in context that can be used in existing Script, ScriptFilter and Validator transforms.
+
+- `CDAP-4403 <https://issues.cask.co/browse/CDAP-4403>`__ -
+  Moved Hydrator plugins from the CDAP repository as cdap-etl-lib into its own repository.
+
+- `CDAP-4022 <https://issues.cask.co/browse/CDAP-4022>`__ -
+  Added RESTful APIs to start or stop multiple programs.
+
+- `CDAP-4023 <https://issues.cask.co/browse/CDAP-4023>`__ -
+  Added CLI commands to stop, start, restart, or get status of programs in an application.
+
+- `CDAP-4285 <https://issues.cask.co/browse/CDAP-4285>`__ -
+  Added an authorization plugin that uses a system dataset to manage ACLs
+
+- `CDAP-4235 <https://issues.cask.co/browse/CDAP-4235>`__ -
+  Added AuthorizationPlugin, for pluggable authorization.
+
+- `CDAP-4591 <https://issues.cask.co/browse/CDAP-4591>`__ -
+  Improved Metadata Indexing and Search to support searches on words in value and tags.
+
+- `CDAP-4592 <https://issues.cask.co/browse/CDAP-4592>`__ -
+  Schema fields are stored as Metadata and are searchable.
+
+- `CDAP-4658<https://issues.cask.co/browse/CDAP-4658>`__ -
+  Added capability in UI to display system tags
+
+Improvements
+------------
+- `CDAP-4209 <https://issues.cask.co/browse/CDAP-4209>`__ -
+  Added a method to ObjectMappedTable and ObjectStore to get a specific number of splits between a start and end key
+
+- `CDAP-4217 <https://issues.cask.co/browse/CDAP-4217>`__ -
+  Deprecated createDataSchedule and createTimeSchedule methods in Schedules class and removed
+  deprecated Schedule constructor.
+
+- `CDAP-3079 <https://issues.cask.co/browse/CDAP-3079>`__ -
+  Table datasets, and any other dataset that implements RecordWritable<StructuredRecord>,
+  can now be written to using Hive.
+
+- `CDAP-4444 <https://issues.cask.co/browse/CDAP-4444>`__ -
+  Make xmllint dependency optional and allow setting variables to skip configuration file parsing
+
+- `CDAP-4133 <https://issues.cask.co/browse/CDAP-4133>`__ -
+  Added ability to get the live-info for the AppFabric system service.
+
+- `CDAP-4233 <https://issues.cask.co/browse/CDAP-4233>`__ -
+  Messages logged by Hydrator are now prefixed with the name of the stage that logged them.
+
+- `CDAP-4453 <https://issues.cask.co/browse/CDAP-4453>`__ -
+  Added Schema validation for sources, transforms and sinks, that will validate the pipeline stages schema during
+  deployment and report issues if any.
+
+- `CDAP-4392 <https://issues.cask.co/browse/CDAP-4392>`__ -
+  Upgraded netty-http dependency in CDAP to 0.14.0
+
+- `CDAP-4536 <https://issues.cask.co/browse/CDAP-4536>`__ -
```

Comment:
Improved installation experience by providing capabilities in CDAP Master service to check pre-requisites such as file system, availability of components such as yarn and hbase, resource availability during startup and error out if any of the pre-requisites fail